File: internal/content/testdata/bench/go/selection-sort-after-k-passes.go
func selection_sort_k_passes(arr []int, k int) []int {
	result := make([]int, len(arr))
	copy(result, arr)
	n := len(result)
	passes := k
	if n-1 < passes {
		passes = n - 1
	}
	for i := 0; i < passes; i++ {
		minIdx := i
		for j := i + 1; j < n; j++ {
			if result[j] < result[minIdx] {
				minIdx = j
			}
		}
		result[i], result[minIdx] = result[minIdx], result[i]
	}
	return result
}

File: internal/content/testdata/bench/rust/selection-sort-after-k-passes.rs
fn selection_sort_k_passes(arr: Vec<i64>, k: i64) -> Vec<i64> {
    let mut result = arr.clone();
    let n = result.len();
    let passes = k.min(n as i64 - 1);
    if passes <= 0 {
        return result;
    }
    let passes = passes as usize;
    for i in 0..passes {
        let mut min_idx = i;
        for j in (i + 1)..n {
            if result[j] < result[min_idx] {
                min_idx = j;
            }
        }
        result.swap(i, min_idx);
    }
    result
}
